### PR TITLE
daemon:fixed an error that the CNI configuration file could not be read

### LINKED ISSUE #26622: A null pointer occurs while reading netConf file
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

A bug happened!
[This operation does not read the CNI configuration file to *netConf
 ](https://github.com/cilium/cilium/blob/c068a443049b5dedb34608069df4b1721b1e4669/daemon/cmd/daemon.go#L420)
In ipamAWS、ipamAlibaba mode，netConf will be empty，and cannot work properly.
image

### Cilium Version

v1.14.0-rc.0"

### Kernel Version

Linux cilium-ipam-test-node 6.4.0-1.el8.elrepo.x86_64 #1 SMP PREEMPT_DYNAMIC Mon Jun 26 14:28:50 EDT 2023 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.4", GitCommit:"f89670c3aa4059d6999cb42e23ccb4f0b9a03979", GitTreeState:"clean", BuildDate:"2023-04-12T12:13:53Z", GoVersion:"go1.19.8", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.4", GitCommit:"f89670c3aa4059d6999cb42e23ccb4f0b9a03979", GitTreeState:"clean", BuildDate:"2023-04-12T12:05:35Z", GoVersion:"go1.19.8", Compiler:"gc", Platform:"linux/amd64"}

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Hawkzilla**
Signed-off-by: bumblebeek <gleemanx@126.com>

Fixes: #26622
We hope fix this bug , which may cause some problems in IPAMAWS、IPAMAlibaba mode.

### CODE REVIEW

**Review by squeed — CHANGES_REQUESTED** [review]
Rather than adding a new function, can we just fix GetMTU to do the right thing?

File: daemon/cmd/cni/config.go [hunk 1]
```diff
@@ -42,29 +42,35 @@ type cniConfigManager struct {
...
+func (c *cniConfigManager) GetNetConf() *cnitypes.NetConf {
 	if c.config.ReadCNIConf == "" {
-		return 0
+		return nil
 	}
 
 	conf, err := cnitypes.ReadNetConf(c.config.ReadCNIConf)
 	if err != nil {
 		c.log.WithField("path", c.config.ReadCNIConf).WithError(err).Warnf("Failed to parse existing CNI configuration file")
 	}
```
Comment (squeed) [inline daemon/cmd/cni/config.go]: there is a bug here in the existing codebase; this should return 0 here as well.

**Comment by bumblebeeK** [comment]
> Rather than adding a new function, can we just fix GetMTU to do the right thing?

okay

**Comment by bumblebeeK** [comment]
> Rather than adding a new function, can we just fix GetMTU to do the right thing?

Do we need to change the GetMTU function signature

**Comment by squeed** [comment]
> Do we need to change the GetMTU function signature ?

I don't think so. But I'm afraid I don't currently understand the bug this is trying to fix.

**Comment by bumblebeeK** [comment]
> don't

https://github.com/cilium/cilium/blob/7726560faf2193a9a7bf06560aec830dcf49414a/daemon/cmd/daemon.go#L519
If not fixed, *netconf is nil in this function

**Comment by thorn3r** [comment]
thanks for the PR, @bumblebeeK! I'm in the same position, I think the bug @squeed pointed out is worth fixing, but I'm not seeing how this fixes the linked issue https://github.com/cilium/cilium/issues/26622

**Comment by bumblebeeK** [comment]
> 

Ok ,thx for your reply, i will do it!

**Pull request closed without merging**